util/include/Metrics.hpp

```diff
@@ -162,13 +165,21 @@ class Component {
   // If registration happens before all registration of the values, then the
   // names will not properly exist in the aggregator, since only values get
   // updated at runtime for performance reasons.
-  void Register() { aggregator_->RegisterComponent(*this); }
+  void Register() {
+    values_.last_aggregated_ = std::chrono::system_clock::now();
+    aggregator_->RegisterComponent(*this);
+  }
 
   void UpdateAggregator() {
+    values_.last_aggregated_ = std::chrono::system_clock::now();
```

Comment:
I would have an internal method to have monotonic wall time, e.g. by having 2 internal variables that are set when the program starts (base wall time and base monotonic time) and then just have currentWallTime = baseWallTime + (currentMonotonicTime - baseMonotonicTime). Does it sounds reasonable? In this way we can have readable monotonic time.

Reply:
Thanks for the review Igor!

I'm not sure that implementation makes sense for our use case. This timestamp is primarily, ( possibly only ever ), going to be used by humans to make sense of the stats they are looking at in real time. The value as proposed will tend to drift over time from the wall clock time, and so it may be confusing for someone looking at the latest time and seeing that it's in the future.

However, if some system is going to be retrieving and recording these responses, monotonic time does make sense. But really what it such a system will want is just to know that one aggregation occurred after another. That also helps with testing, as we can wait for an update that way without checking the values completely. 

How about if we add just a uint64 counter that we increment each time along with the timestamp and always return both as a pair?